api/src/wfl/service/rawls.clj

```diff
@@ -37,14 +38,33 @@
    (create-snapshot-reference workspace snapshot-id name "")))
 
 (defn get-snapshot-reference
-  "Return the snapshot reference in fully-qualified Terra WORKSPACE with REFERENCE-ID."
+  "Return the snapshot reference in `workspace` with `reference-id`."
   [workspace reference-id]
-  (get-workspace-json workspace "snapshots" reference-id))
+  (get-workspace-json workspace snapshot-endpoint reference-id))
+
+(defn get-snapshot-references
+  "Lazily returns the snapshot references in `workspace`
+  with `limit` resources per page (default: 100)."
```

Comment:
I welcome other suggestions for a different page size default.

Reply:
Hard to tell without profiling and real uses. I tend to favour powers of 2 because much two-ness. 100 should be fine.